### PR TITLE
stop database from being deleted on change

### PR DESCRIPTION
**Pull request by jamesrwarren**
## Purpose
To stop the database from being recreated and snapshots lost by terraform on a change. We have added 3 extra flags that should have been there which should stop it happening again!!

These are 
- "database protection" to prevent destruction
- "delete_automated_backups" to false which will stop automated backup being destroyed on destruction of database
- "skip_final_snapshot" to false so that final snapshot will be taken before destruction in the case where we force a destruction.

Fixes DDPB-3469

## Approach
Change the flags in terraform.

## Learning
NA

## Checklist
- [x] I have performed a self-review of my own code
- [ ] I have updated documentation (Confluence/ADR/tech debt doc) where relevant
- [ ] I have added tests to prove my work
- [ ] The product team have approved these changes

### Frontend
- [ ] I have run an in-browser accessibility test (e.g. WAVE, Lighthouse)
- [ ] There are no deprecated CSS classes noted in the profiler
- [ ] Translations are used and the profiler doesn't identify any missing

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -77,9 +77,24 @@ workflows:
           tf_tier: shared
           tf_workspace: development
 
+      - apply:
```
Comment (acsauk-mojd) [inline .circleci/config.yml:80]: I feel like the naming for this job should change now - we're using apply for plan, apply and destroy. Struggling to think of a catchall though that snappy. Something like `run terraform command` is descriptive. I'd also drop the default for apply if you decide to change the job name and be explicit about passing it in when we want to run apply.

File: .circleci/config.yml [hunk 2]
```diff
@@ -128,16 +143,31 @@ workflows:
...
+          filters: { branches: { only: [ master ] } }
+          tf_workspace: production02
+          tf_command: plan
+
+      - apply:
+          name: plan shared-production
+          requires: [ apply preproduction ]
+          filters: { branches: { only: [ master ] } }
+          tf_tier: shared
+          tf_workspace: production
```
Comment (aiesoftware) [inline .circleci/config.yml:158]: `production02`?

File: .circleci/config.yml (hunk 1 above)
Comment (jamesrwarren) [reply .circleci/config.yml:80]: Yeah i did want to do that. I call it terraform action in my other integration pipelines but didn't want to make too many changes and risk borking the pipeline but you're right, let's go for it. Might as well do it now.

File: .circleci/config.yml (hunk 2 above)
Comment (jamesrwarren) [reply .circleci/config.yml:158]: No shared is interestingly in production.. :)

Comment (acsauk-mojd) [reply .circleci/config.yml:158]: Yeah I never got this either 🤷

**Pull request merged**